core/retries/src/main/java/software/amazon/awssdk/retries/internal/LegacyRetryStrategyImpl.java

```diff
@@ -0,0 +1,385 @@
+                    .state(DefaultRetryToken.TokenState.SUCCEEDED)
+                    .build();
+    }
+
+    private void throwOnAcquisitionFailure(RefreshRetryTokenRequest request, AcquireResponse acquireResponse) {
+        DefaultRetryToken token = asStandardRetryToken(request.token());
+        if (acquireResponse.acquisitionFailed()) {
+            Throwable failure = request.failure();
+            DefaultRetryToken refreshedToken =
+                token.toBuilder()
+                     .capacityRemaining(acquireResponse.capacityRemaining())
+                     .capacityAcquired(acquireResponse.capacityAcquired())
+                     .state(DefaultRetryToken.TokenState.TOKEN_ACQUISITION_FAILED)
+                     .addFailure(failure)
+                     .build();
+            String message = acquisitionFailedMessage(acquireResponse);
+            LOG.error(() -> message, failure);
+            throw new TokenAcquisitionFailedException(message, refreshedToken, failure);
+        }
+    }
+
+    private void throwOnMaxAttemptsReached(RefreshRetryTokenRequest request, AcquireResponse acquireResponse) {
+        DefaultRetryToken token = asStandardRetryToken(request.token());
+        if (maxAttemptsReached(token)) {
+            Throwable failure = request.failure();
+            DefaultRetryToken refreshedToken =
+                token.toBuilder()
+                     .capacityRemaining(acquireResponse.capacityRemaining())
+                     .capacityAcquired(acquireResponse.capacityAcquired())
+                     .state(DefaultRetryToken.TokenState.MAX_RETRIES_REACHED)
+                     .addFailure(failure)
+                     .build();
+            String message = maxAttemptsReachedMessage(refreshedToken);
+            LOG.error(() -> message, failure);
+            throw new TokenAcquisitionFailedException(message, refreshedToken, failure);
+        }
+    }
+
+    private void throwOnNonRetryableException(RefreshRetryTokenRequest request, AcquireResponse acquireResponse) {
+        DefaultRetryToken token = asStandardRetryToken(request.token());
+        Throwable failure = request.failure();
+        if (isNonRetryableException(request)) {
+            String message = nonRetryableExceptionMessage(token);
+            LOG.error(() -> message, failure);
+            DefaultRetryToken refreshedToken =
+                token.toBuilder()
+                     .capacityRemaining(acquireResponse.capacityRemaining())
+                     .capacityAcquired(acquireResponse.capacityAcquired())
+                     .state(DefaultRetryToken.TokenState.NON_RETRYABLE_EXCEPTION)
+                     .addFailure(failure)
+                     .build();
+            throw new TokenAcquisitionFailedException(message, refreshedToken, failure);
+        }
+        int attempt = token.attempt();
+        LOG.warn(() -> String.format("Request attempt %d encountered retryable failure.", attempt), failure);
```

Comment:
Same here, should probably be `debug`. I think most of the users use warn as the the root log level and we need to be careful about not polluting their logs.